com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -17,6 +17,31 @@ public sealed class NetworkObject : MonoBehaviour
         [SerializeField]
         internal uint GlobalObjectIdHash;
 
+        /// <summary>
+        /// Gets the Prefab Hash Id of this object if the object is registerd as a prefab otherwise it returns 0
+        /// </summary>
+        [HideInInspector]
+        public uint PrefabIdHash
+        {
+            get
+            {
+                foreach (var prefabList in NetworkManager.NetworkConfig.Prefabs.NetworkPrefabsLists)
```

Comment:
Can we cache the result of this so it doesn't have to do this for-loop every time it's accessed? This seems like it would be pretty expensive, but as a property, it's not obvious that it's expensive to access.

Reply:
I don't think we can unfortunately. I've added some tests and this code isn't correct. There is a Prefabs list in NetworkManager.NetworkConfig.Prefabs.Prefabs which is the actual list of prefabs we need to chefck against. NetworkPrefabsLists is just the list of saved prefabs, any prefabs added at runtime won't be considered.  Since that list can change we can't rely on a cached value.  Fortunately the use case for this is limited to the ConnectionApproval callback so its unlikely its going to get hit very often.  I will update the PR soon with those changes.